### PR TITLE
[sample] disposable handling

### PR DESCRIPTION
**Pull request by lukaszkalnik**
Removes `rxlifecycle` based Disposable handling, leaving only explicit handling in `onPause()`/`onDestroy()`.

Also adds missing `SDK` version check for background scanning calls.

### CODE REVIEW

File: sample/src/main/java/com/polidea/rxandroidble2/sample/example2_connection/ConnectionExampleActivity.java [hunk 1]
```diff
@@ -40,15 +36,15 @@
...
-    private final CompositeDisposable compositeDisposable = new CompositeDisposable();
+    private Disposable mtuDisposable;
+    private Disposable connectionStateDisposable;
 
     @OnClick(R.id.connect_toggle)
     public void onConnectToggleClick() {
         if (isConnected()) {
             triggerDisconnect();
         } else {
             connectionDisposable = bleDevice.establishConnection(autoConnectToggleSwitch.isChecked())
```
Comment (dariuszseweryn) [inline sample/src/main/java/com/polidea/rxandroidble2/sample/example2_connection/ConnectionExampleActivity.java]: This one does not seem to be properly disposed

File: sample/src/main/java/com/polidea/rxandroidble2/sample/example2_connection/ConnectionExampleActivity.java [hunk 2]
```diff
@@ -58,15 +54,12 @@ public void onConnectToggleClick() {
     @TargetApi(Build.VERSION_CODES.LOLLIPOP)
     @OnClick(R.id.set_mtu)
     public void onSetMtu() {
-        final Disposable disposable = bleDevice.establishConnection(false)
+        mtuDisposable = bleDevice.establishConnection(false)
```
Comment (dariuszseweryn) [inline sample/src/main/java/com/polidea/rxandroidble2/sample/example2_connection/ConnectionExampleActivity.java]: It is possible that the `mtuDisposable` will get overwritten if user would tap the button more than once

File: sample/src/main/java/com/polidea/rxandroidble2/sample/example3_discovery/ServiceDiscoveryExampleActivity.java [hunk 3]
```diff
@@ -39,7 +37,6 @@ public void onConnectToggleClick() {
         connectionDisposable = bleDevice.establishConnection(false)
```
Comment (dariuszseweryn) [inline sample/src/main/java/com/polidea/rxandroidble2/sample/example3_discovery/ServiceDiscoveryExampleActivity.java]: May get overwritten due to user clicks

File: sample/src/main/java/com/polidea/rxandroidble2/sample/example2_connection/ConnectionExampleActivity.java (hunk 1 above)
Comment (lukaszkalnik) [reply sample/src/main/java/com/polidea/rxandroidble2/sample/example2_connection/ConnectionExampleActivity.java]: This one does get disposed by calling `triggerDisconnect()` in `onPause()`.

**Comment by dariuszseweryn** [comment]
🚀

**Pull request merged**